### PR TITLE
Feature/2709 muiv5 migation

### PR DESCRIPTION
**Pull request by ektasawant**
Fixes 2709 .

#### Changes proposed in this Pull Request:
- Migrate material UI version from v4

#### Screenshots:
Screenshot 2022-02-15 at 11 30 59 AM

#### To Test:
- run yarn start in mui folder

### CODE REVIEW

File: mui/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## v2.8.1-beta.0 (Not yet published)
```
Comment (emclaug2) [inline mui/CHANGELOG.md:3]: Sounds like this should be 3.0.0

File: mui/demos/showcase/src/index.tsx [hunk 2]
```diff
@@ -2,19 +2,34 @@ import 'react-app-polyfill/ie11';
...
+}
+
 ReactDOM.render(
     <React.StrictMode>
-        <ThemeProvider theme={createMuiTheme(BLUIThemes.blue)}>
-            <CssBaseline />
-            <App />
-        </ThemeProvider>
+        <StyledEngineProvider injectFirst>
+            <ThemeProvider theme={createTheme(adaptV4Theme(BLUIThemes.blue))}>
```
Comment (emclaug2) [inline mui/demos/showcase/src/index.tsx:28]: When our theme is converted over to MUI5, this `adaptV4Theme` wrapper will go away right?

File: mui/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@brightlayer-ui/icons-mui",
-    "version": "2.8.1",
+    "version": "2.8.1-beta.0",
```
Comment (emclaug2) [inline mui/package.json:3]: 3.0.0-beta.0

File: mui/package.json [hunk 4]
```diff
@@ -10,8 +10,8 @@
     "author": "Brightlayer UI <brightlayer-ui@eaton.com>",
     "license": "BSD-3-Clause",
     "peerDependencies": {
-        "@material-ui/core": ">=3.3.1",
-        "@material-ui/icons": ">=3.0.1"
+        "@mui/material": "^5.4.0",
```
Comment (emclaug2) [inline mui/package.json:13]: ^5.0.0 for both of the dependencies here should be fine.

**Review by emclaug2 — COMMENTED** [review]
The changes look great to me but I cannot verify this works because:

1.  I can't get `yarn start` to work on my windows computer since it only works as-is on a mac.
2. I cannot get `yarn` to work for all the demos on my mac presumably because of zscaler issues.

**Comment by emclaug2** [comment]
@ektasawant You mentioned there was a question before this was ready for review?

**Comment by jeffvg** [comment]
looking at this now

**Comment by jeffvg** [comment]
same as @emclaug2 mentioned...
image

**Comment by ektasawant** [comment]
> @ektasawant You mentioned there was a question before this was ready for review?

@emclaug2 in `mui/README.md` line no 6 link is not valid now https://material-ui.com/style/icons/#svg-material-icons . I am not able to verify what should be replace here. I have updated it to `https://mui.com/components/material-icons/#heading-api` can you check whether this link is correct or not?

**Comment by ektasawant** [comment]
> The changes look great to me but I cannot verify this works because:
> 
> 1. I can't get `yarn start` to work on my windows computer since it only works as-is on a mac.
> 2. I cannot get `yarn` to work for all the demos on my mac presumably because of zscaler issues.

I am also not able to run yarn start in mui folder after login to zscaler 😢 .
Screenshot 2022-02-16 at 11 45 19 AM

File: mui/demos/showcase/src/App.tsx [hunk 5]
```diff
@@ -10,12 +10,12 @@ const sorted = meta.icons.sort((a: any, b: any) => (a.filename > b.filename ? 1
...
 export const App: React.FC = () => (
     <>
-        <Grid container spacing={6} style={{ maxWidth: 700, margin: 'auto' }}>
+        <Grid container spacing={2} style={{ maxWidth: 700, margin: 'auto' }}>
             {sorted.map((icon: any) => {
                 //@ts-ignore
                 const Component = Icons[getMuiIconName(icon.filename)];
                 return (
-                    <Grid item xs={2} key={icon.filename}>
+                    <Grid item xs={2} key={icon.filename} style={{ padding: '16px' }}>
```
[the remark this answers is not in the document]
Comment (ektasawant) [reply mui/demos/showcase/src/App.tsx:18]: > 



> While running mui icons, noticed this. Guess we'll need to catch this in themes
> 
> image
> 
> image

Yes it should be fix in react-themes PR.

**Comment by ektasawant** [comment]
> You mentioned there was a question before this was ready for review?

@emclaug2 , please check https://github.com/brightlayer-ui/icons/pull/188#issuecomment-1041399690.

File: mui/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 3.0.0-beta.0 (Not yet published)
```
Comment (joebochill) [inline mui/CHANGELOG.md:3]: We don't list out the alphas and betas in the Changelog. We lump these under what the version is going to be and the "Not Published Yet" message indicates that it is still in pre-release status.

```suggestion
## 3.0.0 (Not yet published)
```

File: mui/README.md [hunk 7]
```diff
@@ -1,10 +1,9 @@
-
 [![](https://img.shields.io/circleci/project/github/brightlayer-ui/icons/mui/master.svg?style=flat)](https://circleci.com/gh/brightlayer-ui/icons/tree/master)
 ![npm](https://img.shields.io/npm/v/@brightlayer-ui/icons-mui?label=%40brightlayer-ui/icons-mui)
 
 # Brightlayer UI Icons for Material-UI
 
-This library contains componentized svg icons from [@brightlayer-ui/icons](https://github.com/brightlayer-ui/icons) for use in React applications using Material UI. The icons are made available in the same way that Material UI exposes the [Material Icons](https://material-ui.com/style/icons/#svg-material-icons).
+This library contains componentized svg icons from [@brightlayer-ui/icons](https://github.com/brightlayer-ui/icons) for use in React applications using Material UI. The icons are made available in the same way that Material UI exposes the [Material Icons](https://mui.com/components/material-icons/#heading-api).
```
Comment (joebochill) [inline mui/README.md:6]: This is the link you want: https://mui.com/components/icons/

**Review by joebochill — COMMENTED** [review]
These changes to package names in the mui-icon-builder folder make me a little nervous. If it works, it's probably fine for now. The longer-term solution is to replace that whole chunk of code with the latest from MUI: https://github.com/mui/material-ui/tree/3914e13d6ac9677f7b57e550b62d3b5f8f830391/packages/mui-icons-material

**Review by daileytj — COMMENTED** [review]
Looks like these console errors are still in here. @jeffvg already called them out so hopefully we have a theme story in place to fix these. Otherwise, see Joe's clean up comments and I think this is good to go. Screen Shot 2022-02-21 at 9 50 55 AM
Screen Shot 2022-02-21 at 9 43 14 AM

**Comment by ektasawant** [comment]
> Looks like these console errors are still in here. @jeffvg already called them out so hopefully we have a theme story in place to fix these. Otherwise, see Joe's clean up comments and I think this is good to go. Screen Shot 2022-02-21 at 9 50 55 AM Screen Shot 2022-02-21 at 9 43 14 AM

we have story for react theme migration which should fix this errors.

**Comment by ektasawant** [comment]
> we're getting lots of notifications about dependencies in that subfolder having vulnerabilities, etc.

How to check this notifications , can you guide me?

**Comment by jeffvg** [comment]
@ektasawant and @joebochill I think Joe is talking about these notifications ...maybe
https://github.com/brightlayer-ui/icons-mui/security/dependabot

**Comment by emclaug2** [comment]
> > You mentioned there was a question before this was ready for review?
> 
> @emclaug2 , please check [#188 (comment)](https://github.com/brightlayer-ui/icons/pull/188#issuecomment-1041399690).

The link you have looks correct to me.

File: mui/demos/showcase/src/App.tsx [hunk 8]
```diff
@@ -1,28 +1,61 @@
...
+    const classes = useStyles(theme);
+
+    return (
+        <>
+            <Grid container spacing={2} classes={{ root: classes.container }}>
+                {sorted.map((icon: any) => {
+                    //@ts-ignore
+                    const Component = Icons[getMuiIconName(icon.filename)];
+                    return (
+                        <Grid item xs={2} key={icon.filename} classes={{ root: classes.gridItem }}>
```
[the remark this answers is not in the document]
Comment (emclaug2) [reply mui/demos/showcase/src/App.tsx:50]: Looks good.

**Review by emclaug2 — APPROVED** [review]
Looks to me that Joe's comments are addressed and we have follow-up stories here to fix any theme issues.

**Review by jeffvg — APPROVED** [review]

**Pull request merged**